pkg/controller/controller_binding.go

```diff
@@ -198,36 +199,84 @@ func (c *controller) reconcileServiceBindingAdd(binding *v1beta1.ServiceBinding)
 		return c.processServiceBindingOperationError(binding, readyCond)
 	}
 
-	if instance.Spec.ClusterServiceClassRef == nil || instance.Spec.ClusterServicePlanRef == nil {
-		// retry later
-		msg := fmt.Sprintf(`Binding cannot begin because ClusterServiceClass and ClusterServicePlan references for %s have not been resolved yet`, pretty.ServiceInstanceName(instance))
-		readyCond := newServiceBindingReadyCondition(v1beta1.ConditionFalse, errorServiceInstanceRefsUnresolved, msg)
-		return c.processServiceBindingOperationError(binding, readyCond)
-	}
+	var prettyName string
+	var brokerClient osb.Client
+	var request *osb.BindRequest
+	var inProgressProperties *v1beta1.ServiceBindingPropertiesState
 
-	serviceClass, servicePlan, brokerName, brokerClient, err := c.getClusterServiceClassPlanAndClusterServiceBrokerForServiceBinding(instance, binding)
-	if err != nil {
-		return c.handleServiceBindingReconciliationError(binding, err)
-	}
+	if instance.Spec.ClusterServiceClassSpecified() {
+		if instance.Spec.ClusterServiceClassRef == nil || instance.Spec.ClusterServicePlanRef == nil {
+			// retry later
+			msg := fmt.Sprintf(`Binding cannot begin because ClusterServiceClass and ClusterServicePlan references for %s have not been resolved yet`, pretty.ServiceInstanceName(instance))
```

Comment:
pretty print the msg?

Reply:
@n3wscott I'm not quite sure what you mean by that? :) not trying to obtuse, but clearly there is a pretty printer used that I missed, is there a good example of it being used elsewhere?